controllers/fenceagentsremediation_controller.go

```diff
@@ -47,8 +46,16 @@ var (
 // FenceAgentsRemediationReconciler reconciles a FenceAgentsRemediation object
 type FenceAgentsRemediationReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log          logr.Logger
+	Scheme       *runtime.Scheme
+	ExecutorHook func(*corev1.Pod) (cli.Executer, error) // for testing the cli command
```

Comment:
I don't think so (at least not the way the code is constructed at the moment).
Since in the production code we are [fetching the executor](https://github.com/medik8s/fence-agents-remediation/blob/main/controllers/fenceagentsremediation_controller.go#L96) in order to mock the executor we need to mock the fetching part.

Reply:
I see. Refactorig the executor would make the code easier than using the func hook though IMHO.
E.g.:  https://github.com/slintes/fence-agents-remediation/commit/b8a9af412e7ecd9150c557a3bdd910615c8aeb6f

Reply:
I agree, I think it simplify the code nicely :+1:

Reply:
Nice idea @slintes !